src/templates/school-nav-search.twig

```diff
@@ -0,0 +1,445 @@
+{% extends "school-layout.twig" %}
+
+{% set page_header = {
+  title: 'School Nav Search',
+  breadcrumb: [
+    { text: 'Offices and Services' },
+    { text: 'Student Financial Services' },
+  ]
+} %}
+
+{% block page %}
+  {% embed 'partials/sidebar-layout.twig' %}
+    {% block nav %}
+
+      {% include 'partials/quicksearch.twig' with {
+        searchElem: 'h2'
+      } %}
+
+      {% include 'partials/page-nav.twig' with {
+        label: 'Student Financial Services',
+        items: [
+          { text: 'Curriculum', href: '#' },
+          {
+            text: 'Faculty',
+            href: '#',
+            items: [
+              { text: 'Faculty sub page 1', href: '#' },
+              { text: 'Faculty sub page 2', href: '#' },
+              {
+                text: 'Faculty sub page 3',
+                href: '#',
+                active: true
+              },
+              { text: 'Faculty sub page 4', href: '#' },
+            ]
+          },
+          { text: 'Financing Your Education', href: '#' },
+        ]
+      } %}
+
+    {% endblock %}
+    {% block toc %}
+
+      <div data-digest-nav data-sticky data-sticky-offset=".js-headroom"></div>
+
+    {% endblock %}
+    {% block content %}
+
+<h2>Afghanistan
```

Comment:
The list of searchable items could be wrapped in a `<div>` with a specific ID or class. This way you can get the searchable items inside that `<div>` itself instead of the entirety of `#midd-main`. The `<h2>` and it's related `<p>` elements can be wrapped in a tag, in this case `<article>` seems to be appropriate. So maybe do something like:
```
<article>
  <h2>Afghanistan</h2>
  <p>Lorem</p>
  <p>..... but also, ipsum.</p>
</article>
```
Then in the JavaScript, you can just do RegExp matching with the article element's text content and that should still work.
The logic in the `findResults` function would be something like:
```
const items = document.querySelectorAll('article');
 
items.forEach((item: HTMLElement) => {
  const allTextContent = item.textContent || '';

  const pattern = new RegExp(`${value}`, 'gi');
  const matches = allTextContent.match(pattern);

  if (matches) {
    matchedItems.push(item);
  }
});
```
This way you can search all of the text content of the item instead of just the h2. Also, you would have to hide or show only one element instead of needing to traverse it's siblings. This way you could have images in the searchable items as well, and the text search _should_ still work.

Reply:
Having a wrapper div would be very helpful, but the markup is dictated by the page which doesn't know whether this component is in the left column or not. I suppose it would be best to change `#midd-main` to `.paragraphs` which is the generic wrapper div for the Basic Content components region.

Reply:
Unfortunately, the markup on the Drupal site is whatever the user enters into a WYSIWYG Text component, so we won't be able to put a wrapper div around each item.

We should update the wrapper div it is searching within from `#midd-main` to `.paragraphs` which is the contains for content on the Basic Content page, but again, that contain will include all items, not single items, so the refined search proposed won't work.

Reply:
Aah okay, there's those constraint. Could it work the way the Events Schedule component works like on the [HR events page](https://www.middlebury.edu/office/human-resources/employee-experiences-and-events), where the schedule items are wrapped in `<div class="schedule-item">`s? 
But I guess that's something that can be done once the requirements for this component are better known.

Reply:
Unfortunately, what we're attempted to do here is create a component that could feasibly be used on any page without having to create a custom component or content type, so it won't include an useful wrapper markup.

I've attempted to update the code to use `.paragraphs` as the container element, but ran into an error I couldn't resolve. That's on the new `473-quicksearch-paragraphs` branch.

Reply:
The error was because it was searching for the elements with classnames `js-quicksearch-form` and `js-quicksearch-input` inside the `.paragraphs` `div`. I [pushed a fix](https://github.com/middlebury/midd-frontend/commit/7f0b45f68eadacc686e2bd185c70929744eb524f) for it where it only looks for the searchable items in the `.paragraphs` `div`.

Reply:
@poojagunturu96  I've added a check to determine if the form element is on the page before proceeding. Does that address your concerns about this running on every page load?

Reply:
Yes, I think this works. Thanks, Ian!